File: backend/rs-client/php/src/service/CreativeService.php
<?php

/**
 
 */
class CreativeService {

    private $configuration;

    public function __construct(RsConfiguration $configuration) {
        $this->configuration = $configuration;
    }

    
    public function get (array $creativeSelector) {
        $client = $this->configuration->createClient();
        $parameters = null;
        $data = null;
    
        $parameters = $creativeSelector;
        
    
        return $client
                ->request("GET" ,"/creatives/", $parameters, $data);
    }
    
    public function perform ($creativeOperations) {
        $client = $this->configuration->createClient();
        $parameters = null;
        $data = null;
    
        
        $data = $creativeOperations;
    
        return $client
                ->request("POST" ,"/creatives/", $parameters, $data);
    }
}

File: backend/rs-client/java/src/main/java/com/foros/rs/client/service/CreativeService.java
package com.foros.rs.client.service;

import com.foros.rs.client.RsClient;
import com.foros.rs.client.data.JAXBEntity;
import com.foros.rs.client.model.advertising.campaign.Creative;
import com.foros.rs.client.model.advertising.campaign.CreativeSelector;
import com.foros.rs.client.model.advertising.template.LivePreviewResult;

public class CreativeService extends EntityServiceSupport<Creative, CreativeSelector> {

    public CreativeService(RsClient rsClient) {
        super(rsClient, "/creatives");
    }

    public LivePreviewResult preview(Long creativeId) {
        return rsClient.get("/creatives/preview?creative.id=" + creativeId);
    }

    public LivePreviewResult livePreview(Creative creative) {
        return rsClient.post("/creatives/livePreview", new JAXBEntity(creative));
    }
}